ui/pages/send/gas-display/gas-display.js

```diff
@@ -339,7 +340,15 @@ export default function GasDisplay({ gasError }) {
                         type="inline"
                         className="confirm-page-container-content__link"
                         onClick={() => {
-                          setShowDepositPopover(true);
+                          openBuyCryptoInPdapp();
+                          trackEvent({
+                            event: EVENT_NAMES.NAV_BUY_BUTTON_CLICKED,
+                            category: EVENT.CATEGORIES.NAVIGATION,
+                            properties: {
+                              location: 'Gas Warning Insufficient Funds',
+                              text: 'Buy',
+                            },
+                          });
```

Comment:
Nit: `trackEvent` inside `openBuyCryptoInPdapp` can save initialising `trackEvent`.

Reply:
Hmmm, guess you could pass the location/text into useRamps if that's the only thing that changes from one invocation to another. But if there are cases that you don't need to track an event i'd say take this into consideration for a future refinement.

Reply:
@brad-decker I thought about doing this change, but then I said keeping the tracking code closer to the point of interaction (trigger) could be more efficient and traceable.